src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -4044,14 +4072,17 @@ public IEnumerator<TElement> GetEnumerator()
         public TElement[] ToArray()
         {
             Buffer<TElement> buffer = new Buffer<TElement>(source);
+
             int count = buffer.count;
-            TElement[] array = new TElement[count];
-            if (count > 0)
+            if (count == 0)
             {
-                int[] map = SortedMap(buffer);
-                for (int i = 0; i != array.Length; i++) array[i] = buffer.items[map[i]];
+                return Array.Empty<TElement>();
```

Comment:
`return buffer.items`? It'll point to the cached empty in this case, and that pointer should be more likely to be in CPU cache, I think.